common/persistence/cassandra/cassandraPersistence.go

```diff
@@ -1109,26 +1081,33 @@ func (d *cassandraPersistence) CreateWorkflowExecution(
 					columns = append(columns, fmt.Sprintf("%s=%v", k, v))
 				}
 
-				if execution, ok := previous["execution"].(map[string]interface{}); ok {
-					// CreateWorkflowExecution failed because it already exists
-					executionInfo := createWorkflowExecutionInfo(execution)
+				if state, ok := previous["execution_state"].([]byte); ok {
+					stateEncoding, ok := previous["execution_state_encoding"].(string)
+					if !ok {
+						return nil, newPersistedTypeMismatchError("execution_state_encoding", "", stateEncoding, previous)
+					}
+
+					protoState, err := serialization.WorkflowExecutionStateFromBlob(state, stateEncoding)
```

Comment:
It will be great if this logic sits in the Manager layer (which takes care of the serialization/deserialization) and does not needs to be implemented by each persistence implementation.

Reply:
You can just put a TODO comment for now to deal with this later.